hudi-common/src/main/java/org/apache/hudi/common/util/ClusteringUtils.java

```diff
@@ -124,7 +125,16 @@
         // get all filegroups in the plan
         getFileGroupEntriesInClusteringPlan(clusteringPlan.getLeft(), clusteringPlan.getRight()));
 
-    Map<HoodieFileGroupId, HoodieInstant> resultMap = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    Map<HoodieFileGroupId, HoodieInstant> resultMap;
+    try {
+      resultMap = resultStream.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    } catch (Exception e) {
+      if (e instanceof IllegalStateException && e.getMessage().contains("Duplicate key")) {
+        throw new HoodieException("Found duplicate file groups pending clustering. If you're running deltastreamer in continuous mode, consider adding delay using --min-sync-interval-seconds. "
```

Comment:
would the in process lock provider help here? Since we then just allow once clustering to be scheduled at once?

Reply:
with OCC mode and in process lock provider, we should not hit this exception.

Reply:
anyway, now we have OCC with in process lock provider when metadata is enabled and users just need to set one config to adjust concurrency mode in case of deltastreamer/spark streaming: `HoodieWriteConfig#AUTO_ADJUST_LOCK_CONFIGS`